noxfile.py

```diff
@@ -32,6 +33,7 @@
 PYO3_DOCS_TARGET = PYO3_TARGET / "doc"
 PY_VERSIONS = ("3.7", "3.8", "3.9", "3.10", "3.11", "3.12", "3.13")
 PYPY_VERSIONS = ("3.7", "3.8", "3.9", "3.10")
+IS_FREE_THREADED = bool(sysconfig.get_config_var("Py_GIL_DISABLED"))
```

Comment:
not sure this is the "correct" way to do this in nox

Reply:
So this makes it conditional on the host python, but really what we care about is the target python, I think?

Reply:
I think as far as our use of `nox` is concerned, these are the same. Is there an alternative?

It looks like the reason this was added is because we made the `abi3` build on free-threaded a hard failure. Maybe we should relax that and then we don't need a change in the `noxfile` for free-threading?